engines/sci/graphics/view.cpp

```diff
@@ -681,39 +655,40 @@ void GfxView::unpackCel(int16 loopNo, int16 celNo, byte *outPtr, uint32 pixelCou
 	}
 }
 
-const byte *GfxView::getBitmap(int16 loopNo, int16 celNo) {
-	loopNo = CLIP<int16>(loopNo, 0, _loopCount -1);
-	celNo = CLIP<int16>(celNo, 0, _loop[loopNo].celCount - 1);
+const SciSpan<const byte> &GfxView::getBitmap(int16 loopNo, int16 celNo) {
+	loopNo = CLIP<int16>(loopNo, 0, _loop.size() - 1);
+	celNo = CLIP<int16>(celNo, 0, _loop[loopNo].cel.size() - 1);
 	if (_loop[loopNo].cel[celNo].rawBitmap)
-		return _loop[loopNo].cel[celNo].rawBitmap;
+		return *_loop[loopNo].cel[celNo].rawBitmap;
 
 	uint16 width = _loop[loopNo].cel[celNo].width;
 	uint16 height = _loop[loopNo].cel[celNo].height;
 	// allocating memory to store cel's bitmap
-	int pixelCount = width * height;
-	_loop[loopNo].cel[celNo].rawBitmap = new byte[pixelCount];
-	byte *pBitmap = _loop[loopNo].cel[celNo].rawBitmap;
+	const uint pixelCount = width * height;
+	_loop[loopNo].cel[celNo].rawBitmap.allocate(pixelCount, _resource->getDescription() + Common::String::format(" loop %d cel %d", loopNo, celNo));
+	SciSpan<byte> rawBitmap = *_loop[loopNo].cel[celNo].rawBitmap;
 
 	// unpack the actual cel bitmap data
-	unpackCel(loopNo, celNo, pBitmap, pixelCount);
+	unpackCel(loopNo, celNo, rawBitmap);
 
 	if (_resMan->getViewType() == kViewEga)
-		unditherBitmap(pBitmap, width, height, _loop[loopNo].cel[celNo].clearKey);
+		unditherBitmap(rawBitmap, width, height, _loop[loopNo].cel[celNo].clearKey);
 
 	// mirroring the cel if needed
 	if (_loop[loopNo].mirrorFlag) {
+		byte *pBitmap = rawBitmap.getUnsafeDataAt(0, width * height);
 		for (int i = 0; i < height; i++, pBitmap += width)
 			for (int j = 0; j < width / 2; j++)
 				SWAP(pBitmap[j], pBitmap[width - j - 1]);
 	}
-	return _loop[loopNo].cel[celNo].rawBitmap;
+	return *_loop[loopNo].cel[celNo].rawBitmap;
 }
 
 /**
  * Called after unpacking an EGA cel, this will try to undither (parts) of the
  * cel if the dithering in here matches dithering used by the current picture.
  */
-void GfxView::unditherBitmap(byte *bitmapPtr, int16 width, int16 height, byte clearKey) {
+void GfxView::unditherBitmap(SciSpan<byte> &bitmapPtr, int16 width, int16 height, byte clearKey) {
```

Comment:
It's not clear to me what the SciSpan adds here, since all accesses go via `getUnsafeDataAt`, and it's a very well-defined "rectangular" block of memory.